libcontainer/cgroups/systemd/unified_hierarchy.go

```diff
@@ -269,23 +259,20 @@ func joinCgroupsV2(c *configs.Cgroup, pid int) error {
 	return createCgroupsv2Path(path)
 }
 
-func (m *UnifiedManager) Freeze(state configs.FreezerState) error {
-	path, err := getSubsystemPath(m.Cgroups, "freezer")
-	if err != nil {
-		return err
-	}
-	prevState := m.Cgroups.Resources.Freezer
-	m.Cgroups.Resources.Freezer = state
-	freezer, err := unifiedSubsystems.Get("freezer")
+func (m *UnifiedManager) fs() (cgroups.Manager, error) {
+	path, err := m.GetUnifiedPath()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	err = freezer.Set(path, m.Cgroups)
+	return fs2.NewManager(m.Cgroups, path, false)
+}
+
+func (m *UnifiedManager) Freeze(state configs.FreezerState) error {
+	fs, err := m.fs()
```

Comment:
Using "fs" as a variable name here is a bit confusing (given "fs" and "fs2" packages...)

Reply:
changed to `fsManager`